src/infra/http/validation/RequiredFieldsValidator.ts

```diff
@@ -0,0 +1,18 @@
+import { Either, left, right } from '@core/logic/Either'
+import { MissingParamError } from './errors/MissingParamError'
+import { Validator } from './Validator'
+
+export class RequiredFieldsValidator<T = any> implements Validator<T> {
+  public validate(data: T): Either<MissingParamError, null> {
+    const fields = Object.getOwnPropertyNames(data)
+    for (const field of fields) {
+      if (data[field] !== false && !data[field]) {
```

Comment:
If think that we should check against `null`, `undefined` or empty value instead of the condition you wrote, so it's more clear for who's reading the code. What do you think?
```js
if (data[field] === null || data[field] === undefined || data[field] === '') {
```

Reply:
Alright, that is a great idea... But just to improve it, we should trim the field, before check if is an empty string, something like this:
```js
if (data[field] === null || data[field] === undefined || data[field].trim() === '')
```
Because once the user send a field with more than one empty space `user.name = '   '`, the validation `data[field] === ''` by itself, wont reject it.

Reply:
That's great!